File: Algorithms/Python/Count_Items_Matching_a_Rule/Faster_Code.py
class Solution:
    def countMatches(self, items: List[List[str]], ruleKey: str, ruleValue: str) -> int:
        count = 0
        ruleIndex = 0 if ruleKey == "type" else 1 if ruleKey == "color" else 2

        for item in items:
            if item[ruleIndex] == ruleValue:
                count += 1

        return count

File: Algorithms/C++/Count_Items_Matching_a_Rule/Faster_Code.cpp
class Solution {
public:
    int countMatches(vector<vector<string>>& items, string ruleKey, string ruleValue) {
        int count = 0;
        int ruleIndex = (ruleKey == "type") ? 0 : (ruleKey == "color") ? 1 : 2;

        for(int i = 0; i < items.size(); i++){
            if(items[i][ruleIndex] == ruleValue){
                count++;
            }
        }

        return count;
    }
};